src/commands/api/request/rest.ts

```diff
@@ -56,30 +62,50 @@ export class Rest extends SfCommand<void> {
 
     const org = flags['target-org'];
     const streamFile = flags['stream-to-file'];
-    const headers = flags.header ? getHeaders(flags.header) : {};
+    const fileOptions = flags.file
+      ? (JSON.parse(fs.readFileSync(flags.file, 'utf8')) as {
+          body?: string;
+          header?: string[];
```

Comment:
this should be `headers` (flag is singular because we expect user to pass it multiple times)